inference-engine/include/ie_plugin_config.hpp

```diff
@@ -299,6 +299,13 @@ DECLARE_CONFIG_VALUE(TUNING_DISABLED);
 DECLARE_CONFIG_VALUE(TUNING_UPDATE);
 DECLARE_CONFIG_VALUE(TUNING_RETUNE);
 
+/**
+ * @brief 
+ *
+ */
+DECLARE_CONFIG_KEY(MYRIAD_MOCK_DEVICE);
```

Comment:
I'm not sure we need such option. What if we just always compile model, even if there is no connected device?

Reply:
Should we compile model even if there are no devices when we use other applications, e.g. benchmark_tool?
I think we should assert this ASAP and not compile the model.
So we need to provide some information into plugin that it can continue compilation with no connections when we use the compiler tool.

Reply:
Anyway, it is reasonable not to use public configs especially in `inference-engine/include/ie_plugin_config.hpp`.

Reply:
> Should we compile model even if there are no devices when we use other applications, e.g. benchmark_tool?
> I think we should assert this ASAP and not compile the model.
> So we need to provide some information into plugin that it can continue compilation with no connections when we use the compiler tool.

I don't see this as strong enough reason for an extra option. I'd prefer less options here.

Reply:
I am looking for a better way to solve this without any options. But for now there is a concern that it will break something inside myriad executor.